modules/doc/content/help/development/VSCode.md

```diff
@@ -0,0 +1,84 @@
+[VSCode website](https://code.visualstudio.com/). The editor has an automatic update system for both the core editor as
+well as the installed extensions.
+
+### First Steps
+
+- On macOS +Start VSCode+ for the first time press +`F1`+ (or +`Cmd-Shift-P`+) and type `Shell Command: Install 'code' command in PATH`. Press enter. This activates the
+  `code` shell command for use from a terminal.
+- On Linux the shell command is automatically installed.
+- +Close VSCode+. From now on we will _only_ start it from the command line using the `code`
+  terminal command ensuring that VSCode sees the full MOOSE build environment.
+
+## Important commands
+
+- +`Cmd-Shift-P`+ opens the command palette. Every available Atom command can be accessed by typing a
+  few letters here. The dropdown list shows the keyboard shortcuts.
+- +`Cmd-P`+ opens a file anywhere in the current project tree (i.e. below the directory in which you
+  issued the `atom .` command. No need to know the precise path or even the precise spelling of the
+  filename!
+
+## Extensions
+
+
+The following extensions should be installed to effectively develop MOOSE based codes and edit MOOSE
+input files using Atom
+
+- [MOOSE Language Support](https://marketplace.visualstudio.com/items?itemName=DanielSchwen.moose-language-support): Syntax highlighting, automatic
+  indentation, document outlines, mouse hover info, formatting, and auto-completion for MOOSE input files, C++ code snippets for all MOOSE systems, and
+  highlighting of select MOOSE C++ types.
+- [C/C++ Extension Pack](https://marketplace.visualstudio.com/items?itemName=ms-vscode.cpptools-extension-pack) installs a set of extensions to support C++ code navigation, syntax highlighting, etc.
+
+## Recommended settings
+
+The following settings are recommended to pass the MOOSE GitHub CI pre-checks:
+
+```
+{
+    "workbench.colorTheme": "Atom One Dark",
+    "gitlens.hovers.currentLine.over": "line",
+    "[cpp]": {
+        "editor.defaultFormatter": "ms-vscode.cpptools"
+    },
+    "telemetry.telemetryLevel": "off",
+    "C_Cpp.default.cppStandard": "c++17",
+    "C_Cpp.default.includePath": [
+        "${workspaceFolder}/**",
+        "${env.CONDA_PREFIX}/**",
+        "/usr/include/**"
+    ],
+    "workbench.editorAssociations": {
+        "*.C": "cpp",
+        "*.h": "cpp"
+    },
+    "files.trimTrailingWhitespace": true,
+    "editor.tabSize": 2,
+    "editor.formatOnSave": true,
+    "files.insertFinalNewline": true
+}
+```
+
+The `editor.formatOnSave` , `files.insertFinalNewline` , and  `files.trimTrailingWhitespace` options are helpful to avoid getting a failing precheck. The `C_Cpp.default.includePath` settings ensure that includes installed with Conda/Mamba are found. The _Atom One Dark_ theme is available as an extension and provides teh familiar Atom colors. [GitLens](https://marketplace.visualstudio.com/items?itemName=eamodio.gitlens) is an extension that provides extensive git integration.
```

Comment:
```suggestion
The `editor.formatOnSave` , `files.insertFinalNewline` , and  `files.trimTrailingWhitespace` options are helpful to avoid failing the automated code check in MOOSE's test suite. The `C_Cpp.default.includePath` settings ensure that includes installed with Conda/Mamba are found. The _Atom One Dark_ theme is available as an extension and provides the familiar Atom colors. [GitLens](https://marketplace.visualstudio.com/items?itemName=eamodio.gitlens) is an extension that provides extensive git integration.
```